src/array.js

```diff
@@ -6,9 +6,13 @@ if (!Array.from) {
     var toLength = function(it) {
       return it > 0 ? Math.min(toInteger(it), 0x1fffffffffffff) : 0; // pow(2, 53) - 1 == 9007199254740991
     };
-    var iterCall = function(iter, fn, a1, a2) {
+    var iterCall = function(fn, val, index, thisArg) {
       try {
-        fn(a1, a2);
+        if (thisArg) {
+          return fn.call(thisArg, val, index)
+        } else {
+          return fn(val, index);
+        }
       }
       catch (E) {
         if (typeof iter.return == 'function') iter.return();
```

Comment:
To be honest, I don't know what the heck `iter` even is :) . The original code wasn't passing an iterator in, and the iterator function that's being looped through doesn't have a return on it anyway. And this function isn't being called anywhere else either. I'll pass it back through though, as I missed that anyway.